prebuilt-tasks/src/main/java/com/redhat/parodos/tasks/git/GitArchive.java

```diff
@@ -0,0 +1,91 @@
+package com.redhat.parodos.tasks.git;
+
+import com.google.common.io.Files;
+import com.redhat.parodos.workflow.parameter.WorkParameter;
+import com.redhat.parodos.workflow.parameter.WorkParameterType;
+import com.redhat.parodos.workflow.task.BaseWorkFlowTask;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.eclipse.jgit.api.ArchiveCommand;
+import org.eclipse.jgit.archive.ZipFormat;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.io.FileOutputStream;
+
+@AllArgsConstructor
+public class GitArchive extends BaseWorkFlowTask {
+
+	@Autowired
+	private GitUtils gitUtils;
+
+	@Override
+	public @NonNull List<WorkParameter> getWorkFlowTaskParameters() {
+		return List.of(WorkParameter.builder().key(this.gitUtils.getGitRepoPath()).type(WorkParameterType.TEXT)
+				.optional(true).description("path where the git repo is located").build());
+	}
+
+	@Override
+	public WorkReport execute(WorkContext workContext) {
+		String path = this.gitUtils.getRepoPath(workContext);
+		if (path == null || path == "") {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("No valid path: " + path));
+		}
+
+		Repository repo = this.getRepo(path);
+		if (repo == null) {
+			return new DefaultWorkReport(WorkStatus.FAILED, workContext, new Exception("No repository at " + path));
+		}
+
+		try {
```

Comment:
It is a shame the Repository doesn't implement the Closeable interface, else you could replace with try-with-resouce.

pls move the rep.close() to finally block

Reply:
done